arrays/arrays.c

```diff
@@ -21,39 +20,48 @@ typedef struct Array {
  *****/
 Array *create_array (int capacity) {
   // Allocate memory for the Array struct
-
+  Array *arr = malloc(sizeof(Array));
   // Set initial values for capacity and count
-
+  arr->capacity = capacity;
+  arr->count = 0;
   // Allocate memory for elements
-
+  arr->elements = calloc(capacity, sizeof(char *));
+  return arr;
 }
 
 
 /*****
  * Free memory for an array and all of its stored elements
  *****/
 void destroy_array(Array *arr) {
-
   // Free all elements
-
+  if (arr->elements != NULL) {
+    free(arr->elements);
+  }
   // Free array
-
+  if (arr != NULL) {
+    free(arr);
+  }
 }
 
 /*****
  * Create a new elements array with double capacity and copy elements
  * from old to new
  *****/
 void resize_array(Array *arr) {
-
+  
   // Create a new element storage with double capacity
-
+  int newarr = arr->capacity * 2;
   // Copy elements into the new storage
-
+  char **newelements = calloc(newarr, sizeof(char *));
+  for (int i = 0; i < arr->capacity; i++) {
+    newelements[i] = arr->elements[i];
+  }
   // Free the old elements array (but NOT the strings they point to)
-
+  free(arr);
```

Comment:
When I run this it gives me an error. You should be freeing the elements array here, but you're actually freeing the entire arr struct. Just a small syntax error I think. Your code functions perfectly otherwise.